pkg/container/kubernetes/client.go

```diff
@@ -1043,68 +1041,57 @@ func configureContainer(
 		WithTTY(false).
 		WithEnv(envVars...)
 
-	// Add container security context if not already present
+	// Add container security context using SecurityContextBuilder
 	if container.SecurityContext == nil {
-		container.WithSecurityContext(
-			corev1apply.SecurityContext().
-				WithPrivileged(false).
-				WithRunAsNonRoot(true).
-				WithAllowPrivilegeEscalation(false).
-				WithReadOnlyRootFilesystem(true).
-				WithRunAsUser(int64(1000)).
-				WithRunAsGroup(int64(1000)),
-		)
+		securityBuilder := NewSecurityContextBuilder(platform)
```

Comment:
nit for readability: cine the builder and applying the config is needed for both if and else could we put the declaration earlier?

Reply:
Addressed as per the updated [client.go#L1045](https://github.com/RoddieKieley/toolhive/blob/f17cb679abb75cfadb78218f98be2313efe9cb4d/pkg/container/kubernetes/client.go#L1045)